api/cpp/include/slint.h

```diff
@@ -347,6 +358,14 @@ class Window
         return inner.set_rendering_notifier(std::forward<F>(callback));
     }
 
+    /// This function allows registering a callback that's invoked when the user tries to close
+    /// a window.
+    /// The callback has to return a CloseRequestResponse.
+    template<typename F> void on_close_requested(F &&callback) const
+    {
+        return inner.on_close_requested(std::forward<F>(callback));
```

Comment:
Not 100% sure that this compiles, but it might be worth verifying the return type with a good error message.

```suggestion
        static_assert(std::is_invocable_v<Functor>, "Functor callback must be callable");
        static_assert(std::is_same_v<std::invoke_result_t<F>, CloseRequestResponse>, "Functor callback must return CloseRequestResponse");
        return inner.on_close_requested(std::forward<F>(callback));
```